R/tidyeval.R

```diff
@@ -133,8 +133,9 @@ dt_squash_call <- function(x, env, data, j = TRUE) {
     x[[1]] <- sym("-")
     x[[2]] <- get_expr(x[[2]])
     x
-  } else if (is_call(x, "if_else") || is_call(x, "ifelse")) {
+  } else if (is_call(x, c("if_else", "ifelse"))) {
     x[[1L]] <- quote(fifelse)
+    names(x) <- if_else_args[names(x)]
```

Comment:
I wonder if it's worth using `match.call()` here — the small advantage is that it would protect against partial naming; the bigger disadvantage is that it would turn unnamed arguments to named arguments.

Reply:
Maybe use `match.call()` and then remove names from unnamed arguments?

``` r
library(rlang)

if_else_args <- c(
  "condition" = "test", "true" = "yes",
  "false" = "no", "missing" = "na"
)

x <- expr(if_else(x < 1, tr = 1, 0))

if (is_call(x, c("if_else", "ifelse"))) {
  named_flag <- have_name(x)
  if (any(named_flag)) {
    if (is_call(x, "if_else")) {
      x <- match.call(dplyr::if_else, x)
      names(x) <- if_else_args[names(x)]
    } else {
      x <- match.call(base::ifelse, x)
    }
    names(x)[!named_flag] <- ""
  }
  
  x[[1]] <- quote(fifelse)
  x
}
#> fifelse(x < 1, yes = 1, 0)
```

Reply:
I don't think that works because `match.call()` also changes the argument order:

``` r
x <- rlang::expr(if_else(x < 1, 0, tr = 1))
match.call(dplyr::if_else, x)
#> if_else(condition = x < 1, true = 1, false = 0)
```

How about just unconditionally stripping the argument names in the generated call?

Reply:
Sounds good - seems like a good tradeoff